facebook/facebook-api.c

```diff
@@ -922,56 +922,61 @@ fb_api_cb_mqtt_open(FbMqtt *mqtt, gpointer data)
         ? fb_api_get_agent_string(priv->tweak, 1)
         : FB_API_MQTT_AGENT);
 
-    /* Write the UNKNOWN ("cp"?) */
+    /* Write the client capabilities */
     fb_thrift_write_field(thft, FB_THRIFT_TYPE_I64, 3, 2);
-    fb_thrift_write_i64(thft, 23);
+    fb_thrift_write_i64(thft, FB_CP_ACKNOWLEDGED_DELIVERY |
+                              FB_CP_PROCESSING_LASTACTIVE_PRESENCEINFO |
+                              FB_CP_EXACT_KEEPALIVE |
+                              FB_CP_DELTA_SENT_MESSAGE_ENABLED);
 
-    /* Write the UNKNOWN ("ecp"?) */
+    /* Write the endpoint capabilitites */
     fb_thrift_write_field(thft, FB_THRIFT_TYPE_I64, 4, 3);
     fb_thrift_write_i64(thft, 26);
 
-    /* Write the UNKNOWN */
+    /* Write the publish payload format (deflate) */
     fb_thrift_write_field(thft, FB_THRIFT_TYPE_I32, 5, 4);
     fb_thrift_write_i32(thft, 1);
 
-    /* Write the UNKNOWN ("no_auto_fg"?) */
+    /* Write the noAutomaticForeground flag */
     fb_thrift_write_field(thft, FB_THRIFT_TYPE_BOOL, 6, 5);
     fb_thrift_write_bool(thft, TRUE);
 
-    /* Write the visibility state */
+    /* Write the visibility state (makeUserAvailableInForeground flag) */
     fb_thrift_write_field(thft, FB_THRIFT_TYPE_BOOL, 7, 6);
     fb_thrift_write_bool(thft, !priv->invisible);
 
     /* Write the device identifier */
     fb_thrift_write_field(thft, FB_THRIFT_TYPE_STRING, 8, 7);
     fb_thrift_write_str(thft, priv->did);
 
-    /* Write the UNKNOWN ("fg"?) */
+    /* Write the isInitiallyForeground flag */
     fb_thrift_write_field(thft, FB_THRIFT_TYPE_BOOL, 9, 8);
     fb_thrift_write_bool(thft, TRUE);
 
-    /* Write the UNKNOWN ("nwt"?) */
+    /* Write the network type (WIFI) */
     fb_thrift_write_field(thft, FB_THRIFT_TYPE_I32, 10, 9);
     fb_thrift_write_i32(thft, 1);
 
-    /* Write the UNKNOWN ("nwst"?) */
+    /* Write the network subtype (none) */
     fb_thrift_write_field(thft, FB_THRIFT_TYPE_I32, 11, 10);
     fb_thrift_write_i32(thft, 0);
 
     /* Write the MQTT identifier */
     fb_thrift_write_field(thft, FB_THRIFT_TYPE_I64, 12, 11);
     fb_thrift_write_i64(thft, priv->mid);
 
-    /* Write the UNKNOWN */
+    /* Write the list of topics to subscribe */
     fb_thrift_write_field(thft, FB_THRIFT_TYPE_LIST, 14, 12);
     fb_thrift_write_list(thft, FB_THRIFT_TYPE_I32, 0);
+
+    /* Write the STOP for the struct */
     fb_thrift_write_stop(thft);
 
     /* Write the token */
-    fb_thrift_write_field(thft, FB_THRIFT_TYPE_STRING, 15, 14);
+    fb_thrift_write_field(thft, FB_THRIFT_TYPE_STRING, 5, 4);
```

Comment:
`clientInfo` struct has stopped the line above, so you need to continue numeration from the level above. Basically this change does nothing with result, because delta between fields is still `1`, it's just for match the definition.

Here is a definition for `CONNECT` thrift for the reference:
```thrift
struct ClientInfo {
  1:  i64       userId,
  2:  string    userAgent,
  3:  i64       clientCapabilities,
  4:  i64       endpointCapabilities,
  5:  i32       publishFormat,
  6:  bool      noAutomaticForeground,
  7:  bool      makeUserAvailableInForeground,
  8:  string    deviceId,
  9:  bool      isInitiallyForeground,
  10: i32       networkType,
  11: i32       networkSubtype,
  12: i32       clientMqttSessionId,
  13: string    clientIpAddress,
  14: list<i32> subscribeTopics,
  15: string    clientType,
  16: i64       appId,
  17: bool      overrideNectarLogging,
  18: string    connectTokenHash,
  19: string    regionPreference,
  20: string    deviceSecret,
  21: i8        clientStack,
  22: i64       fbnsConnectionKey,
  23: string    fbnsConnectionSecret,
  24: string    fbnsDeviceId,
  24: string    fbnsDeviceSecret
}

struct Connect {
  1: string     clientIdentifier,
  2: string     willTopic,
  3: string     willMessage,
  4: ClientInfo clientInfo,
  5: string     password 
}
```